src/hotspot/cpu/riscv/templateTable_riscv_64.cpp

```diff
@@ -121,21 +118,25 @@ void TemplateTable::patch_bytecode(Bytecodes::Code new_bc, Register Rnew_bc, Reg
       __ get_cache_and_index_at_bcp(Rtemp /* dst = cache */, 1);
       // ((*(cache+indices))>>((1+byte_no)*8))&0xFF:
 #if defined(VM_LITTLE_ENDIAN)
-      __ lbz_PPC(Rnew_bc, in_bytes(ConstantPoolCache::base_offset() + ConstantPoolCacheEntry::indices_offset()) + 1 + byte_no, Rtemp);
+      __ lbu(Rnew_bc, Rtemp, in_bytes(ConstantPoolCache::base_offset() + ConstantPoolCacheEntry::indices_offset()) + 1 + byte_no);
 #else
-      __ lbz_PPC(Rnew_bc, in_bytes(ConstantPoolCache::base_offset() + ConstantPoolCacheEntry::indices_offset()) + 7 - (1 + byte_no), Rtemp);
+      __ lbu(Rnew_bc, Rtemp, in_bytes(ConstantPoolCache::base_offset() + ConstantPoolCacheEntry::indices_offset()) + 7 - (1 + byte_no));
 #endif
-      __ cmpwi_PPC(CCR0, Rnew_bc, 0);
-      __ li_PPC(Rnew_bc, (unsigned int)(unsigned char)new_bc);
-      __ beq_PPC(CCR0, L_patch_done);
-      // __ isync_PPC(); // acquire not needed
+      __ beqz(Rnew_bc, L_zero);
+      __ li(Rnew_bc, (unsigned int)(unsigned char)new_bc);
+      __ sb(Rnew_bc, R22_bcp, 0);
+      __ j(L_patch_done);
```

Comment:
In the original code you jump to the done label only if Rnew_bc is zero. Now you do this in each case

Reply:
Works as intended

Reply:
You skip code in `if (JvmtiExport::can_post_breakpoint()) {`